zen_queries/decorators.py

```diff
@@ -6,20 +6,26 @@ class QueriesDisabledError(Exception):
     pass
 
 
-def _fake(*args, **kwargs):
+def _raise_exception(*args, **kwargs):
     raise QueriesDisabledError()
 
 
 def _apply_monkeypatch(connection):
-    connection._queries_disabled = True
-    connection._real_create_cursor = connection.create_cursor
-    connection.create_cursor = _fake
+    if hasattr(connection, "execute_wrappers"):
+        connection.execute_wrappers.append(_raise_exception)
+    else:
+        connection._queries_disabled = True
+        connection._real_create_cursor = connection.create_cursor
+        connection.create_cursor = _raise_exception
 
 
 def _remove_monkeypatch(connection):
-    connection.create_cursor = connection._real_create_cursor
-    del connection._real_create_cursor
-    del connection._queries_disabled
+    if hasattr(connection, "execute_wrappers"):
+        connection.execute_wrappers.pop()
```

Comment:
If `execute_wrappers` is a plain list, it might be better to explicitly remove our wrapper, rather that removing whichever was last applied. 

```suggestion
        connection.execute_wrappers.remove(_raise_exception)
```

Reply:
I was just mirroring the code in Django itself: https://github.com/django/django/blob/master/django/db/backends/base/base.py#L655-L665